packages/frontend/src/pages/index.tsx

```diff
@@ -8,26 +8,46 @@ import { useCallback, useEffect, useState } from 'react'
 import Image from 'next/image'
 import { multiaddr } from '@multiformats/multiaddr'
 import { connectToMultiaddr } from '../lib/libp2p'
-
+import { useListenAddressesContext } from '../context/listen-addresses-ctx'
 
 export default function Home() {
   const { libp2p } = useLibp2pContext()
   const { peerStats, setPeerStats } = usePeerContext()
+  const { listenAddresses, setListenAddresses } = useListenAddressesContext()
   const [maddr, setMultiaddr] = useState('')
 
   useEffect(() => {
-    const peerConnectedCB = (evt: any) => {
-      const connection = evt.detail
-      setPeerStats({ ...peerStats, peerIds: [...peerStats.peerIds, connection.remotePeer], connections: [...peerStats.connections, connection], connected: true })
-    }
+    const interval = setInterval(() => {
+      const connections = libp2p.getConnections()
 
-    libp2p.addEventListener('peer:connect', peerConnectedCB)
+      setPeerStats({
+        ...peerStats,
+        peerIds: connections.map(conn => conn.remotePeer),
+        connections: connections,
+        connected: true
```

Comment:
In testing I realize this needs to get reset to false if we lose connections, previously this would connect to the rust peer and still connected indefinitely so this issue never arose